include/graaflib/algorithm/cycle_detection.tpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <algorithm>
+#include <unordered_map>
+
+namespace graaf::algorithm {
+
+namespace detail {
+
+enum vertex_color{WHITE, GREY, BLACK};
```

Comment:
[minor]: Could you please provide a short comment to clarify these enum values? Or maybe change the names to something like:
```suggestion

enum vertex_color{UNVISITED, VISITED, NO_CYCLE};
```

Reply:
Hmm, ill add comments when for graph coloring